tests/testthat/test-formatting.R

```diff
@@ -112,3 +112,136 @@ test_that("format_na_str functionality works in get_formatted_cells (ie printing
     expect_identical(get_formatted_cells(tbl)[3, 1, drop = TRUE],
                      "Ridiculous")
 })
+
+test_that("Test matrix obtained from get_formatted_cells as is into ASCII format", {
+
+  # Table builder with rtables
+  tbl <- basic_table() %>%
```

Comment:
We can be more efficient with our tests here. 

If we use an afun such as

```
weird_afun <- function(x, ...) {
in_rows(cell_fmt = rcell(mean(x, na.rm = TRUE), format = "xx.x"),
              no_cell_fmt = rcell(median(x, na.rm = TRUE)),
              cell_na_str = rcell(NA, format = "xx.x", na_str = "haha"),
              no_cell_fmt = rcell(NA))
}
```

And then do 

```
DM2 <- DM
DM2$AGE[1] <- NA
tbl <- basic_table() %>%
    split_cols_by("ARM") %>%
    analyze("AGE", weird_afun, format = "xx.xx", na_str = "woah") %>%
    build_table(DM2)
``` 

Then we can check all four cases against that single matrix_form.

Reply:
```
Error in analyze(., "AGE", weird_afun, format = "xx.xx", na_str = "woah") :
unused argument (na_str = "woah")
```
Also calling the docs this parameter is not present. ps: I tried to find this functionality since the beginning btw, that is why I used `make_afun` in the first place. Also tried unsuccessfully: `extra_args = list(na_str = "woah")`

Reply:
I adapted the test but still no solution for the above comment

Reply:
Yes, na_str is less pervasive than I had thought. You can currently only set it within in_rows (which is what `make_afun` does, so that won't be any different than the "haha" case above). Neither Table nor Row based classes currently have a slot for it, only `CellValue` and `RowsVerticalSection`.

For now leave that off of the test for na_str inheritence, and we will add it back in when we implement that functionality more widely.

Reply:
ok perfect!